code/game/machinery/machinery.dm

```diff
@@ -391,10 +389,12 @@ Class Procs:
 	return 0
 
 /obj/machinery/proc/exchange_parts(mob/user, obj/item/weapon/storage/part_replacer/W)
-	if(flags & NODECONSTRUCT)
+	if(!istype(W))
+		return
+	if(flags & NODECONSTRUCT && !W.works_from_distance)
```

Comment:
Allows BSRPED to change parts when standing next to a NODECONSTRUCT machine, for example a freezer which is in use. It can already do this if you don't stand next to it so basically a bug fix.